app/main/views/direct_award.py

```diff
@@ -201,7 +201,60 @@ def get_project_search(project_id, search_id):
 
 @main.route('/direct-award/projects/<int:project_id>/services', methods=['GET'])
 def list_project_services(project_id):
-    raise NotImplementedError()
+    """This endpoint returns all the services associated with a particular (locked) Direct Award Project. It returns
+    some fairly arbitrary data which is obviously not ideal, but as we don't have a task runner that we can utilise to
+    manage jobs like this and we need it to be responsive on-click, we're having to jerryrig in some data extraction and
+    transformation in here. Specifically we are returning supplier name and contact information, and returning from
+    the service data JSON keys as specified in the request (which will be loaded via a framework-specific manifest).
+
+    TODO: If/When we get a DM task runner, this should be updated to just return a list of service IDs and the runner
+    becomes responsible for querying the API to get information on those services."""
+    page = get_valid_page_or_1()
+    requested_fields = request.args.get('fields', '').split(',')
+    project = DirectAwardProject.query.filter(DirectAwardProject.id == project_id).first_or_404()
+
+    if not project.locked_at:
+        abort(400, 'Project has not been locked: {}'.format(project_id))
+
+    search = DirectAwardSearch.query.filter(
+        DirectAwardSearch.project_id == project_id,
+        DirectAwardSearch.active == True  # noqa
+    ).first()
+    if not search:
+        abort(400, 'Project does not have a saved search: {}'.format(project_id))
+
+    paginated_archived_services = search.archived_services.paginate(
+        page=page,
+        per_page=current_app.config['DM_API_PROJECTS_PAGE_SIZE'],
+    )
+
+    project_archived_services = list(map(lambda service: {
+        'id': service.service_id,
+        'supplier': {
+            'name': service.supplier.name,
+            'contact': {
+                'name': service.supplier.contact_information[0].contact_name,
+                'phone': service.supplier.contact_information[0].phone_number,
+                'email': service.supplier.contact_information[0].email,
+            },
+        },
+        'data': drop_all_other_fields(service.data, requested_fields),
```

Comment:
I'm not sure I entirely hold with this "recursively wipe fields named/not-named X regardless of where they are in the structure" thing. Feels like we should know the structure, and wipe the specific things from where they are. Otherwise in a world of properly-nested related objects, random bits of other objects get wiped if the names clash. But I get that this is what we already do, so eh.

Reply:
This call doesn't recurse, otherwise yes that wouldn't be good.